### PR TITLE
Charts: publish shift method

### PR DESCRIPTION
**Pull request by marker-dao**

### CODE REVIEW

File: js/viz/chart.d.ts [hunk 1]
```diff
@@ -227,6 +227,13 @@ export interface baseLabelObject {
      * @public
      */
     getBoundingRect(): any;
+    /**
+     * @docid
+     * @publicName shift()
```
Comment (IlyaKhD) [inline js/viz/chart.d.ts]:
```suggestion
     * @publicName shift(x, y)
```

**Review by IlyaKhD — DISMISSED** [review]

File: js/viz/chart.d.ts (hunk 1 above)
Comment (marker-dao) [reply js/viz/chart.d.ts]: Thanks! It was updated in [2937b8cbfc93647add7f379babfb6991785c5f94].

**Review by IlyaKhD — DISMISSED** [review]

File: js/viz/chart.d.ts (hunk 1 above)
Comment (IlyaKhD) [reply js/viz/chart.d.ts]: 👌🏻

**Review by Krijovnick — DISMISSED** [review]

File: js/viz/chart.d.ts [hunk 2]
```diff
@@ -227,6 +227,13 @@ export interface baseLabelObject {
      * @public
      */
     getBoundingRect(): any;
+    /**
+     * @docid
+     * @publicName shift(x, y)
+     * @return object
+     * @public
+     */
+    shift(x: number, y: number): any;
```
Comment (alexanderPolosatov) [inline js/viz/chart.d.ts]: method returns _baseLabelObject_ . see:  https://github.com/DevExpress/DevExtreme/blob/23_1/js/viz/series/points/label.js#L514
May be we concretize this type here?

Comment (marker-dao) [reply js/viz/chart.d.ts]: Yes, I agree with it. Fixed here: [755efe07d4738719c9f1e1727edd9e166bdad066].

**Review by alexanderPolosatov — DISMISSED** [review]

**Review by Krijovnick — DISMISSED** [review]

File: js/viz/chart.d.ts [hunk 3]
```diff
@@ -227,6 +227,13 @@ export interface baseLabelObject {
      * @public
      */
     getBoundingRect(): any;
+    /**
+     * @docid
+     * @publicName shift(x, y)
+     * @return object
```
Comment (ksercs) [inline js/viz/chart.d.ts:233]: shouldn't we change `object` to a `baseLabelObject` ?

File: testing/tests/DevExpress.viz.core.series/label.tests.js [hunk 4]
```diff
@@ -1184,19 +1184,35 @@ QUnit.module('Layouted label', $.extend({}, environment, {
...
 
     const innerGroup = label._insideGroup;
+
     assert.ok(innerGroup);
-    assert.deepEqual(innerGroup._stored_settings, { translateX: 10 - 1, translateY: 10 - 2 });
     assert.equal(label._insideGroup.stub('append').lastCall.args[0], label._group);
     assert.equal(label._group.stub('append').lastCall.args[0], this.group);
 });
 
+QUnit.test('Label should be moved', function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.viz.core.series/label.tests.js]: or it's specified in a module name?
```suggestion
QUnit.test('shift method call should move a label on specified delta', function(assert) {
```

File: js/viz/chart.d.ts (hunk 3 above)
Comment (marker-dao) [reply js/viz/chart.d.ts:233]: I have not seen this in other descriptions. Let's ask @IlyaKhD

File: testing/tests/DevExpress.viz.core.series/label.tests.js (hunk 4 above)
Comment (marker-dao) [reply testing/tests/DevExpress.viz.core.series/label.tests.js]: Was fixed here: [49a79514db22d1cf7c7e60d512cae2bb1320f314].

**Review by Zedwag — DISMISSED** [review]

**Review by Krijovnick — DISMISSED** [review]

File: js/viz/chart.d.ts (hunk 3 above)
Comment (IlyaKhD) [reply js/viz/chart.d.ts:233]: No, let's leave it anonymous object for now. `baseLabelObject` is not a public type, having wrong name. It should be reviewed first.

File: js/viz/chart.d.ts [hunk 5]
```diff
@@ -227,6 +227,13 @@ export interface baseLabelObject {
...
+     * @public
+     */
+    shift(x: number, y: number): baseLabelObject;
```
Comment (IlyaKhD) [inline js/viz/chart.d.ts]: **Minior**
```suggestion
    shift(x: number, y: number): this;
```
Works in the same way, but looks ~more stylish~ clearer

**Review by IlyaKhD — DISMISSED** [review]

**Review by IlyaKhD — DISMISSED** [review]

**Review by EugeniyKiyashko — DISMISSED** [review]

**Review by EugeniyKiyashko — APPROVED** [review]

**Review by Krijovnick — APPROVED** [review]

**Review by Zedwag — APPROVED** [review]

**Review by vladaskorohodova — APPROVED** [review]

**Review by IlyaKhD — APPROVED** [review]

**Pull request merged**